handler/src/main/java/io/netty/handler/ssl/JdkAlpnSslEngine.java

```diff
@@ -62,12 +63,15 @@ private static void updateAvailability() {
                     "protocolSelector");
             ALPN.put(engine, new ServerProvider() {
                 @Override
-                public String select(List<String> protocols) {
+                public String select(List<String> protocols) throws SSLException {
                     try {
                         return protocolSelector.select(protocols);
+                    } catch(SSLException e) {
+                        throw e;
                     } catch (Throwable t) {
-                        PlatformDependent.throwException(t);
-                        return null;
+                        // Ensure that all exceptions are propagated as SSLExceptions
+                        // so that the SslHandler properly fails the handshake.
+                        throw new SSLException(t.getMessage());
```

Comment:
I was using the `String` constructor before because I was originally going to throw a `SSLHandshakeException` which doesn't take a throwable in the constructor.  I've changed it to `throw new SSLException(t);` ... any reason not to use this constructor?